x-pack/plugins/uptime/public/pages/monitor.tsx

```diff
@@ -15,6 +15,14 @@ import { useMonitorId, useUptimeTelemetry, UptimePage } from '../hooks';
 import { MonitorCharts } from '../components/monitor';
 import { MonitorStatusDetails, PingList } from '../components/monitor';
 import { getDynamicSettings } from '../state/actions/dynamic_settings';
+import { Ping } from '../../common/runtime_types/ping';
+
+const getIdToDisplay = (monId: string, selectedMonitor: Ping | null) => {
+  if (monId.startsWith('auto-') && monId.length > 20) {
```

Comment:
I almost feel like we should document this; it seems unlikely that people will give their monitors IDs that correspond to these criteria, but if they did it might be confusing when they see the URL instead.

That said, I suppose it's a fairly limited concern.

Reply:
This is a good point. It's something I threw out as an option. Ideally we'd have a special field in heartbeat indicating whether it's auto generated I suppose, OTOH, the values are sufficiently strange that I don't mind doing this detection in this way. I'd be a bit more specific and match a precise regex: `/^auto-(icmp|http|tcp)-0X[A-F0-9]{16}/` for this specific use case.

I've made a new issue that recommends requiring IDs in 8.0: https://github.com/elastic/uptime/issues/189 , comment there would be welcome.

Reply:
I knew we were planning on adding that change in 8.x, just wanted to make sure we discussed the implementation of this here. It might be worth tracking that we remove this function after we can guarantee a custom ID so we lower the risk of strange behavior from special id values.

Reply:
Well, we might decide to have 8.x still handle 7.x data. Currently there are no breaking changes, so it still might make sense to keep the logic here.

I realize that this smells, but given the extreme uniqueness of the ID format, I'm OK with it.

I agree we should comment this well though, explaining exactly what's happening, and probably making a utility function `isAutogeneratedId(id: string)` to wrap this in.

Reply:
@andrewvc shouldn't regex be `^auto-(icmp|http|tcp)-OX[A-F0-9]{16}-[a-f0-9]{16}` ?